src/hypervisor.c

```diff
@@ -106,13 +112,12 @@ cc_oci_expand_net_device_cmdline(struct cc_oci_config *config, guint index) {
 
 	g_debug("PCI Offset used for network: %d", PCI_OFFSET);
 
-	if ( if_cfg->mac_address == NULL ) {
-		return g_strdup_printf(QEMU_FMT_DEVICE,
-			index + PCI_OFFSET,
-			if_cfg->tap_device);
+	if ( if_cfg->mac_address == NULL || ! *if_cfg->mac_address) {
```

Comment:
Could be if ( !if_cfg->mac_address || ! *if_cfg->mac_address) instead. Just to get this shorter.